ee/tabby-db/src/password_reset.rs

```diff
@@ -0,0 +1,55 @@
+use anyhow::Result;
+use chrono::{DateTime, Duration, Utc};
+use sqlx::{prelude::FromRow, query};
+use uuid::Uuid;
+
+use crate::DbConn;
+
+#[derive(FromRow)]
+pub struct PasswordResetDAO {
+    pub user_id: i32,
+    pub code: String,
+    pub created_at: DateTime<Utc>,
+}
+
+impl DbConn {
+    pub async fn create_password_reset(&self, id: i32) -> Result<String> {
+        let code = Uuid::new_v4().to_string();
+        let time = Utc::now();
+        query!(
+            "INSERT INTO password_reset (user_id, code, created_at) VALUES ($1, $2, $3)
+            ON CONFLICT(user_id) DO UPDATE SET code= $2, created_at = $3;",
+            id,
+            code,
+            time
+        )
+        .execute(&self.pool)
+        .await?;
+        Ok(code)
+    }
+
+    pub async fn delete_password_reset(&self, id: String) -> Result<()> {
```

Comment:
```suggestion
    pub async fn delete_password_reset_by_user_id(&self, user_id: String) -> Result<()> {
```

please also update methods that's `by_user` id with such suffix.